extensions/grpc/deployment/src/main/java/io/quarkus/grpc/deployment/GrpcServerProcessor.java

```diff
@@ -690,7 +690,9 @@ ServiceStartBuildItem initializeServer(GrpcServerRecorder recorder,
             LaunchModeBuildItem launchModeBuildItem,
             VertxWebRouterBuildItem routerBuildItem,
             VertxBuildItem vertx, Capabilities capabilities,
-            List<FilterBuildItem> filterBuildItems) {
+            List<FilterBuildItem> filterBuildItems,
+            // used to ensure that gRPC server starts after OTel has been set up
+            @SuppressWarnings("unused") BeanContainerBuildItem beanContainerBuildItem) {
```

Comment:
What about if we're more specific / exact, to "wait" for OTel, not just entire ArC?
e.g. to add some new build item ...

Reply:
I tried one of the BuildItem classes we already have in OTel and unfortunately it didn't work because although gRPC server started after OTel, the OTel bean is lazy and still caused the same problem